mmengine/analysis/print_helper.py

```diff
@@ -696,6 +697,13 @@ def get_model_complexity_info(
     Returns:
         dict: The complexity information of the model.
     """
+    assert (input_shape is not None) or (
+        inputs
+        is not None), '"input_shape" and "inputs" cannot both be "None".'
+    if (input_shape is not None) and (inputs is not None):
+        warnings.warn('Both "input_shape" and "inputs" are provided.\
+            "inputs" will be used.')
```

Comment:
```suggestion
    if input_shape is None and inputs is None:
        raise ValueError(
            'One of the input_shape and inputs should be set.')
    elif input_shape is not None and inputs is not None:
        raise ValueError(
            'input_shape and inputs can not be both set.')
```

Reply:
About the error message, how about

```python
if input_shape is None and inputs is None:
    raise ValueError(
        'One of input_shape and inputs should be set.')
# The rest suggestion is fine
```

When neither "input_shape" nor "inputs" is provided, the error message should guide the user to provide one of them. The expression "Only one of input_shape and inputs can be set." is suitable for the scenario that one sets both "input_shape" and "inputs".

Furthermore, I suggest using double quotes as delimiter for "input_shape" and "inputs". That is

```python
if input_shape is None and inputs is None:
    raise ValueError(
        'One of "input_shape" and "inputs" should be set.')
# The rest suggestion is fine
```

as so in https://github.com/open-mmlab/mmengine/blob/1c67f9eb22c5528266c2748d146429c2c7336e19/mmengine/fileio/io.py#L836-L841

Reply:
I completely agree. 👍